### PR TITLE
Convert integration tests to use `lit`

### PR DESCRIPTION
**Pull request by Baltoli**
These changes mirror the ones made to the C semantics:
* Add LLVM's `lit` package to the Dockerfile so that it's installed and available for the CI runner. This uses the same steps as the C semantics to install `lit`.
* Convert the tests in `defn` and `unparse` to be run as individual `lit` cases, rather than from root Makefiles.
* Small changes to the CI runner script to deal with files being generated into different directories.
* Add an example `envrc.sh` as we have done for other projects; this handles setting your paths appropriately.

### CODE REVIEW

**Comment by Baltoli** [comment]
A note on converting some of the tests in `defn` - they appear not to finish rewriting (exit code 113). They still produce output on `stdout` that matches their reference output; my assumption is that this behaviour is intentional, but I wanted to check that was the case. The solution is adding `%allow-pipefail` to allow the interpreter execution to fail.

**Comment by Baltoli** [comment]
This is blocked until I can bump the version of `lit` that nixpgs references: https://github.com/NixOS/nixpkgs/blob/a0dbe47318bbab7559ffbfa7c4872a517833409f/pkgs/development/tools/misc/lit/default.nix

**Comment by Baltoli** [comment]
@dwightguth This is ready for review now that I've fixed the Nix issues from last week.

File: ciscript [hunk 1]
```diff
@@ -37,18 +37,14 @@ cd ../build
...
-cp ../test/Makefile ./TestMakefile
+TEST_DIR=../test
 
-make -f TestMakefile -j`nproc`
+lit $TEST_DIR
 
-diff io_output1.txt ../test/ioTest1.output
-diff `ls | grep io_log` ../test/ioTest3.output
+cd $TEST_DIR/defn
+diff io_output1.txt ../ioTest1.output
```
Comment (dwightguth) [inline ciscript:45]: These checks can probably be inlined into the respective test now.

File: test/lit.cfg.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+config.name = 'llvm-backend'
+config.test_source_root = os.path.join(ROOT_PATH, "test")
+
+config.test_format = formats.ShTest(True)
+
+config.suffixes = ['.kore']
+
+config.excludes = []
+
+if os.getenv('LIT_USE_NIX'):
```
Comment (dwightguth) [inline test/lit.cfg.py:16]: Can we get a comment here about what this is doing?

File: test/unparse/exp/lit.local.cfg.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+config.excludes = [
```
Comment (dwightguth) [inline test/unparse/exp/lit.local.cfg.py:1]: Should we move these into the parent directory somehow since it's the same across each directory?

**Review by dwightguth — CHANGES_REQUESTED** [review]
Overall this looks good, but I'd like to see a few minor changes

File: ciscript (hunk 1 above)
Comment (Baltoli) [reply ciscript:45]: Will do.

File: test/lit.cfg.py (hunk 2 above)
Comment (Baltoli) [reply test/lit.cfg.py:16]: Will comment it - the problem is basically `lit` not inheriting the environment when it's launched, which of course breaks nix derivations.

**Comment by Baltoli** [comment]
@dwightguth The manual section of `ciscript` is now inlined (with cleanup) into the run scripts for `test-io` and `test28` respectively.

**Review by dwightguth — APPROVED** [review]
Great work, thanks!

**Pull request merged**